File: frontend/src/Models/Order.ts
export interface OrderGet {
  id: number;
  createdOn: string;
  paymentMethod: string;
  amount: number;
  status: number;
  appUserName: string;
  shippingAddress : string;
}

export interface OrderPost {
  paymentMethod: string;
  amount: number;
  shippingAddress : string;
}

File: api/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace api.Models
{
    public class Order
    {
        public int Id { get; set; }
        public DateTime CreatedOn { get; set; } = DateTime.Now;
        public string PaymentMethod { get; set; } = string.Empty;

        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }
        public int Status { get; set; }
        public string ShippingAddress { get; set; } = string.Empty;
        public string AppUserId { get; set; }
        public AppUser AppUser { get; set; }


    }
}